src/Common/Auth/OpenIDConnect/Repositories/ScopeRepository.php

```diff
@@ -61,9 +74,16 @@ public function getScopeEntityByIdentifier($scopeIdentifier)
             'nonce' => [
                 'description' => 'Security',
             ],
+            'patient/*.read' => [
+                'description' => 'Read only access to all information about a patient that currently exists and any information created in the future.',
+            ],
+            'launch/patient' => [
+                'description' => 'Grant an external application the ability to launch and load your patient profile.',
+            ]
```

Comment:
these also need to get into client registration, claims etc.
I've already have broken out the claims for smart or oidc. As I do client grant i'm also adding the system/ scopes/claims etc.
So i'm pulling in your changes/commits so we don't have too many conflicts and i'll try to get my client grant minus dealing with having to handshake the client JWK for client signature verification up tonight while you sleep:). I don't have a client to test with as yet. This is for backend stuff but will also fixup dynamic register for smart.

Reply:
@sjpadgett There's a bunch of apps you can download and use to test against the OpenEMR stuff here [https://github.com/smart-on-fhir/sample-apps](https://github.com/smart-on-fhir/sample-apps )

I'll see if I can fix my style changes so you don't have to clean that up.

I've been using the Medications Standalone app.  Just have to edit launch.html  Here's an example:
`
FHIR.oauth2.authorize({
          "client": {
              "client_id": "j-tJl9Rl0fQssGnHUhDtvumo1LEcxFdJyWk5WAKLxqM",
              "scope":  "patient/*.read launch/patient"
          },
          //"server": "https://launch.smarthealthit.org/v/r2/sim/eyJoIjoiMSIsImoiOiIxIn0/fhir"
          "server": "https://localhost:9300/apis/default/fhir"
      });
`

Reply:
cool thanks. Don't worry about the styling errors. I'll fix, don't mind.

Reply:
@adunsulag what grant is this using for auth?
FHIR.oauth2.authorize({ "client": { "client_id": "j-tJl9Rl0fQssGnHUhDtvumo1LEcxFdJyWk5WAKLxqM", "scope": "patient/*.read launch/patient" }, //"server": "https://launch.smarthealthit.org/v/r2/sim/eyJoIjoiMSIsImoiOiIxIn0/fhir" "server": "https://localhost:9300/apis/default/fhir" });

Reply:
It uses the Authorization Code grant

Reply:
I had to preregister a client the redirect_URIs needs to be http://localhost:9090/  I spent 30 minutes figuring out why my oauth was being rejected.  The browser sends the trailing slash.  That was the motivation for me to add more logging to the oauth code.

Reply:
That reminds me. Are you using the registered global domain for pathing. Can't use the server to grab domain type urls due to container/AWS and proxies etc. Thought I saw that somewhere you were building paths.

Kudos on logging. Brady will be happy.